1.19.0-alpha.3

matteobruni opened the pull request:
fix: orbit color is now a string/color and not a hsl object
chore: updated editor with new options
feat: all h/s/l values can now be animated
fix: fixed all hsl animations with the right offset value
feat: added core build, just the skeleton, no plugins added
refactor: ColorUtils is becoming obsolete, the new function structure will be the future
feat: particle initial overlap can be now configured

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.8.1 to 4.8.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.8.2/packages/eslint-plugin)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @babel/preset-typescript from 7.12.1 to 7.12.7

Bumps [@babel/preset-typescript](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-typescript) from 7.12.1 to 7.12.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.7/packages/babel-preset-typescript)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.1.9 to 8.1.10

Bumps [postcss](https://github.com/postcss/postcss) from 8.1.9 to 8.1.10.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.1.9...8.1.10)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ng-packagr from 11.0.2 to 11.0.3

Bumps [ng-packagr](https://github.com/ng-packagr/ng-packagr) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/ng-packagr/ng-packagr/releases)
- [Changelog](https://github.com/ng-packagr/ng-packagr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ng-packagr/ng-packagr/compare/v11.0.2...v11.0.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/webpack-env from 1.15.3 to 1.16.0

Bumps [@types/webpack-env](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/webpack-env) from 1.15.3 to 1.16.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/webpack-env)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1048 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.8.2

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.8.1 to 4.8.2

matteobruni committed:
Merge pull request #1052 from matteobruni/dependabot/npm_and_yarn/dev/types/webpack-env-1.16.0

build(deps-dev): bump @types/webpack-env from 1.15.3 to 1.16.0

matteobruni committed:
Merge pull request #1049 from matteobruni/dependabot/npm_and_yarn/dev/babel/preset-typescript-7.12.7

build(deps-dev): bump @babel/preset-typescript from 7.12.1 to 7.12.7

matteobruni committed:
Merge pull request #1050 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.1.10

build(deps-dev): bump postcss from 8.1.9 to 8.1.10

matteobruni committed:
Merge pull request #1051 from matteobruni/dependabot/npm_and_yarn/dev/ng-packagr-11.0.3

build(deps-dev): bump ng-packagr from 11.0.2 to 11.0.3

matteobruni committed:
fix: orbit color is now a string/color and not a hsl object
refactor: sorted some properties in options

ghost commented:
## DeepCode's analysis on [#cef85d](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ec8e2ab4909dba45f3eade716a7e9f725e4de414/matteobruni/tsparticles/cef85dd781ce4a0f62256401d38db99b268a3d9d/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=3&) found:

- :information_source: **3** minor issues. :point_down:
- :heavy_check_mark: **3** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/Options/Classes/HslAnimation.ts#L7">HslAnimation.ts:7</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/Options/Classes/HslAnimation.ts#L8">HslAnimation.ts:8</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/Options/Classes/HslAnimation.ts#L9">HslAnimation.ts:9</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ec8e2ab4909dba45f3eade716a7e9f725e4de414/matteobruni/tsparticles/cef85dd781ce4a0f62256401d38db99b268a3d9d/pr/_/%2Fcore%2Fmain%2Fsrc%2FOptions%2FClasses%2FHslAnimation.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Expected property shorthand in object literal ('{afterEffect}'). Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/main.core.ts#L143">main.core.ts:143</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/main.core.ts#L144">main.core.ts:144</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/main.core.ts#L146">main.core.ts:146</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ec8e2ab4909dba45f3eade716a7e9f725e4de414/matteobruni/tsparticles/cef85dd781ce4a0f62256401d38db99b268a3d9d/pr/_/%2Fcore%2Fmain%2Fsrc%2Fmain.core.ts/javascript%2Ftslint%2Fobject-literal-shorthand/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/cef85dd781ce4a0f62256401d38db99b268a3d9d/core/main/src/main.core.ts#L50">main.core.ts:50</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ec8e2ab4909dba45f3eade716a7e9f725e4de414/matteobruni/tsparticles/cef85dd781ce4a0f62256401d38db99b268a3d9d/pr/_/%2Fcore%2Fmain%2Fsrc%2Fmain.core.ts/javascript%2Ftslint%2Farray-type/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/ec8e2ab4909dba45f3eade716a7e9f725e4de414/matteobruni/tsparticles/cef85dd781ce4a0f62256401d38db99b268a3d9d/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=3&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
build: changed one undefined property syntax

matteobruni committed:
chore: updated editor with new options

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 4.8.1 to 4.8.2

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.8.1 to 4.8.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.8.2/packages/parser)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump jsoneditor from 9.1.3 to 9.1.4

Bumps [jsoneditor](https://github.com/josdejong/jsoneditor) from 9.1.3 to 9.1.4.
- [Release notes](https://github.com/josdejong/jsoneditor/releases)
- [Changelog](https://github.com/josdejong/jsoneditor/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/jsoneditor/compare/v9.1.3...v9.1.4)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump uglify-js from 3.11.6 to 3.12.0

Bumps [uglify-js](https://github.com/mishoo/UglifyJS) from 3.11.6 to 3.12.0.
- [Release notes](https://github.com/mishoo/UglifyJS/releases)
- [Commits](https://github.com/mishoo/UglifyJS/compare/v3.11.6...v3.12.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump typescript-json-schema from 0.43.0 to 0.44.0

Bumps [typescript-json-schema](https://github.com/YousefED/typescript-json-schema) from 0.43.0 to 0.44.0.
- [Release notes](https://github.com/YousefED/typescript-json-schema/releases)
- [Commits](https://github.com/YousefED/typescript-json-schema/compare/v0.43.0...v0.44.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1054 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/parser-4.8.2

build(deps-dev): bump @typescript-eslint/parser from 4.8.1 to 4.8.2

matteobruni committed:
Merge pull request #1055 from matteobruni/dependabot/npm_and_yarn/dev/jsoneditor-9.1.4

build(deps-dev): bump jsoneditor from 9.1.3 to 9.1.4

matteobruni committed:
Merge pull request #1057 from matteobruni/dependabot/npm_and_yarn/dev/uglify-js-3.12.0

build(deps-dev): bump uglify-js from 3.11.6 to 3.12.0

matteobruni committed:
Merge pull request #1058 from matteobruni/dependabot/npm_and_yarn/dev/typescript-json-schema-0.44.0

build(deps-dev): bump typescript-json-schema from 0.43.0 to 0.44.0

matteobruni committed:
build: updated svelte rollup plugin and fixed build errors

matteobruni committed:
build: deps updated

matteobruni committed:
docs: updated move docs

matteobruni committed:
feat: all h/s/l values can now be animated

matteobruni committed:
fix: fixed all hsl animations with the right offset value

matteobruni committed:
build: updated deps

dependabot[bot] committed:
build(deps-dev): bump @types/react-dom from 16.9.9 to 17.0.0

Bumps [@types/react-dom](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react-dom) from 16.9.9 to 17.0.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react-dom)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1063 from matteobruni/dependabot/npm_and_yarn/dev/types/react-dom-17.0.0

build(deps-dev): bump @types/react-dom from 16.9.9 to 17.0.0

matteobruni committed:
build: updated deps

matteobruni committed:
docs: updated color animation docs

matteobruni committed:
feat: added core build, just the skeleton, no plugins added
refactor: ColorUtils is becoming obsolete, the new function structure will be the future

matteobruni committed:
build: updated colorutils test

matteobruni committed:
refactor: removed image shape from Particle.ts

matteobruni committed:
build: fixed imports

matteobruni committed:
build: renamed unbreaking property to unbreakable

dependabot[bot] committed:
build(deps-dev): bump @types/node from 14.14.9 to 14.14.10

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 14.14.9 to 14.14.10.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump react-syntax-highlighter from 15.3.0 to 15.3.1

Bumps [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) from 15.3.0 to 15.3.1.
- [Release notes](https://github.com/react-syntax-highlighter/react-syntax-highlighter/releases)
- [Changelog](https://github.com/react-syntax-highlighter/react-syntax-highlighter/blob/master/CHANGELOG.MD)
- [Commits](https://github.com/react-syntax-highlighter/react-syntax-highlighter/compare/v15.3.0...v15.3.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump typescript-json-schema from 0.44.0 to 0.44.1

Bumps [typescript-json-schema](https://github.com/YousefED/typescript-json-schema) from 0.44.0 to 0.44.1.
- [Release notes](https://github.com/YousefED/typescript-json-schema/releases)
- [Commits](https://github.com/YousefED/typescript-json-schema/compare/v0.44.0...v0.44.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump svelte from 3.29.7 to 3.30.0

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.29.7 to 3.30.0.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.29.7...v3.30.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.6.0 to 5.8.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.6.0 to 5.8.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.6.0...v5.8.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1064 from matteobruni/dependabot/npm_and_yarn/dev/types/node-14.14.10

build(deps-dev): bump @types/node from 14.14.9 to 14.14.10

matteobruni committed:
Merge pull request #1065 from matteobruni/dependabot/npm_and_yarn/dev/react-syntax-highlighter-15.3.1

build(deps-dev): bump react-syntax-highlighter from 15.3.0 to 15.3.1

matteobruni committed:
Merge pull request #1066 from matteobruni/dependabot/npm_and_yarn/dev/typescript-json-schema-0.44.1

build(deps-dev): bump typescript-json-schema from 0.44.0 to 0.44.1

matteobruni committed:
Merge pull request #1067 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.30.0

build(deps): bump svelte from 3.29.7 to 3.30.0

matteobruni committed:
Merge pull request #1068 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.8.0

build(deps-dev): bump webpack from 5.6.0 to 5.8.0

matteobruni committed:
feat: particle initial overlap can be now configured (closes #1069)

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
fix: fixed initial position with new implementations made for #1069

matteobruni committed:
refactor: simplified recursive calcPosition call

matteobruni committed:
docs: updated docs for #1069

matteobruni committed:
chore: updated editor to latest collisions properties

matteobruni committed:
build: checked sea anemone preset before new alpha release

matteobruni committed:
chore(release): published new version

 - ng-particles@2.2.0-alpha.3
 - inferno-particles@1.2.0-alpha.3
 - jquery-particles@1.19.0-alpha.3
 - preact-particles@1.19.0-alpha.3
 - react-tsparticles@1.19.0-alpha.3
 - svelte-particles@1.2.0-alpha.3
 - particles.vue@2.2.0-alpha.3
 - particles.vue3@1.2.0-alpha.3
 - tsparticles-editor@1.19.0-alpha.3
 - tsparticles@1.19.0-alpha.3
 - ng-particles-demo@1.4.0-alpha.3
 - editor-demo@1.4.0-alpha.3
 - inferno-particles-demo@1.2.0-alpha.3
 - jquery-particles-demo@1.4.0-alpha.3
 - tsparticles-demo@1.4.0-alpha.3
 - preact-particles-demo@1.4.0-alpha.3
 - react-tsparticles-demo@1.4.0-alpha.3
 - svelte-demo@1.2.0-alpha.3
 - particles.vue-demo@1.4.0-alpha.3
 - particles.vue3-demo@1.2.0-alpha.3
 - tsparticles-preset-big-circles@1.4.0-alpha.3
 - tsparticles-preset-fire@1.4.0-alpha.3
 - tsparticles-preset-fountain@1.4.0-alpha.3
 - tsparticles-preset-links@1.4.0-alpha.3
 - tsparticles-preset-sea-anemone@1.4.0-alpha.3
 - tsparticles-preset-snow@1.4.0-alpha.3
 - tsparticles-preset-stars@1.4.0-alpha.3
 - tsparticles-shape-bubble@1.4.0-alpha.3
 - tsparticles-shape-heart@1.4.0-alpha.3
 - tsparticles-shape-multiline-text@1.4.0-alpha.3
 - tsparticles-shape-spiral@1.4.0-alpha.3
 - cra-template-particles-typescript@1.2.0-alpha.3
 - cra-template-particles@1.2.0-alpha.3
